controllers/machinehealthcheck_controller_test.go

```diff
@@ -46,6 +46,71 @@ import (
 
 const defaultNamespaceName = "default"
 
+func newMachine(mhc *clusterv1.MachineHealthCheck, cluster *clusterv1.Cluster) clusterv1.Machine {
+	return clusterv1.Machine{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: clusterv1.GroupVersion.String(),
+			Kind:       "Machine",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-mhc-machine-",
+			Namespace:    mhc.Namespace,
+			Labels:       mhc.Spec.Selector.MatchLabels,
+		},
+		Spec: clusterv1.MachineSpec{
+			ClusterName: cluster.Name,
+			Bootstrap: clusterv1.Bootstrap{
+				Data:           pointer.StringPtr("data"),
+				DataSecretName: pointer.StringPtr("data-secret-name"),
+			},
+		},
+		Status: clusterv1.MachineStatus{
+			InfrastructureReady: true,
+			BootstrapReady:      true,
+			Phase:               string(clusterv1.MachinePhaseRunning),
+			ObservedGeneration:  1,
+		},
+	}
+}
+
+func newNode(healthy bool) corev1.Node {
```

Comment:
It can be confusing to know what the bool is for when you see a call like `newNode(true)` vs `newNode(false)`. You don't necessarily need to change this as it's test code, but you might consider making this just `newNode()` that defaults to Ready, and adding another helper function like `setNodeUnhealthy(node *corev1.Node)` that modifies the condition.